public/javascripts/app.js

```diff
@@ -0,0 +1,101 @@
+import { handleErrors } from "./utils.js";
+
+const fetchTasks = async () => {
+    const res = await fetch("http://localhost:8080/tasks")
+
+    if (res.status === 401) {
+        window.location.href = "/log-in";
+        return;
+      }
+
+    const { tasks } = await res.json();
+    const tasksListContainer = document.querySelector(".task-list");
+    const tasksHtml = tasks.map(({ id, description }) => `
+    <div>
+        <input type="checkbox" class="task-check-box" id=${id} name=${id}>
+        <label for=${id} class="task-check-box">${description}</label>
+    </div>
+    `)
+
+    tasksListContainer.innerHTML = tasksHtml.join("");
+}
+
+document.addEventListener("DOMContentLoaded", async () => {
+    try {
+      await fetchTasks();
+    } catch (e) {
+      console.error(e);
+    }
+  });
+
+const form = document.querySelector(".create-task");
+
+form.addEventListener("submit", async (e) => {
+    e.preventDefault();
+    const formData = new FormData(form);
+    const description = formData.get("description")
+    const dueDate = formData.get("dueDate")
+    const checkStatus = formData.get("isCompleted")
+    const givenTo = formData.get("givenTo")
+    let isCompleted;
+    let csrf;
+
+    //convert checkbox to boolean value
+    if (checkStatus === 'on') {
+       isCompleted = true;
+    } else {
+        isCompleted = false;
+    }
+
```

Comment:
This parse-ing out the csrf token shouldn't be necessary if it's included in the form field as a hidden value.